src/ansys/dynamicreporting/core/utils/report_utils.py

```diff
@@ -51,20 +48,209 @@ def encode_url(s):
     return s
 
 
-def is_enve_image(img):
-    if has_enve and has_qt:  # pragma: no cover
-        return isinstance(img, enve.image)
-    return False
+def check_if_PIL(img):
+    """
+    Check if the input image can be opened by PIL.
+
+    Parameters
+    ----------
+    img:
+        filename or bytes representing the picture
+
+    Returns
+    -------
+    bool:
+        True if the image can be opened by PIL
+    """
+    # Assume you are getting bytes.
+    # If string, open it
+    imgbytes = img
+    if isinstance(img, str):
+        imgbytes = open(img, "rb")
+    try:
+        # Check PIL can handle the img opening
+        Image.open(imgbytes)
+        return True
+    except Exception:
+        return False
+    finally:
+        imgbytes.close()
```

Comment:
@mariostieriansys If imgbytes is bytes, then this will fail.
I think what you want to do is check that img is str, and in that case you've opened the file at line 69 and now want to close it. But it won't work as it is now.

Reply:
I think my issue here is the docstring says imgbytes can be a byte type, while really it should be more of a _io.TextIOWrapper type. So I guess changing the doc string is enough.

Reply:
@margalva
I was trying to cover this use case:

with open("example.png", 'rb') as fp:
        img = fp.read()
item.set_payload_image(img)
        
      problem is, in the code I have been treating it as an instance of _io.TextIOWrapper but in the docstrings they are bytes. They should be bytes and I should change the code accordingly

Reply:
@margalva I pushed some changes. It should be clearer now.
The use cases, according to the documentation, are that you can either pass a path, the bytes read from a file, or the bytes of an enve image.
There are now two variables defaulted to None, where only one gets populated, depending on the input type. If a file handle is generated during the execution, is closed at the end